src/main/webapp/feed.html

```diff
@@ -25,21 +25,126 @@ <h1>
+    <select onchange="addFilter()" id="tag-select">
+      <option>None</option>
+    </select>
+
+    <p>Currently selected filters:</p>
+    <div id="selected-filters"></div>
+    <br />
+
     <ul id="recipe-list"></ul>
 
     <script>
       let recipeList = document.getElementById("recipe-list")
-      fetch("/api/post").then(async (response) => {
-        if (response.status < 200 || response.status >= 300) {
-          alert("Error " + response.status.toString())
-        } else {
-          let recipes = await response.json()
-          recipes.forEach((recipe) => {
-            recipeList.innerHTML +=
-              '<li><a href="recipe-display.html?recipeID=' + recipe.id + '">' + recipe.title + "</a></li>"
-          })
+      let tagSelect = document.getElementById("tag-select")
+
+      window.onload = function () {
+        // Fetch all recipes to populate our list
+        fetch("/api/post").then(async (response) => {
+          if (response.status < 200 || response.status >= 300) {
+            alert("Error " + response.status.toString())
+          } else {
+            let recipes = await response.json()
+            recipes.forEach((recipe) => {
+              recipeList.innerHTML +=
+                '<li><a href="recipe-display.html?recipeID=' + recipe.id + '">' + recipe.title + "</a></li>"
+            })
+          }
+        })
+
+        // Fetch all recipe tags to populate our select form
+        fetch("/api/tag").then(async (response) => {
+          if (response.status < 200 || response.status >= 300) {
+            alert("Error " + response.status.toString())
+          } else {
+            let tags = await response.json()
+            tags.forEach((tag) => {
+              tagSelect.innerHTML += '<option value="' + tag.id + '">' + tag.name + "</option>"
+            })
+          }
+        })
+      }
+
+      // Performs retrieval of recipes based on the filters currently contained in <div> 'selected-filters'
+      function getRecipes() {
+        const tagIDs = getCurrentFilters()
+        let recipeList = document.getElementById("recipe-list")
+        fetch("/api/post?tagIDs=" + tagIDs).then(async (response) => {
+          if (response.status < 200 || response.status >= 300) {
+            alert("Error " + response.status.toString())
+          } else {
+            let recipes = await response.json()
+            recipeList.innerHTML = ""
+            recipes.forEach((recipe) => {
+              recipeList.innerHTML +=
+                '<li><a href="recipe-display.html?recipeID=' + recipe.id + '">' + recipe.title + "</a></li>"
+            })
+          }
+        })
+      }
+
+      function getCurrentFilters() {
+        var filterString = ""
+        var filters = document.getElementById("selected-filters").childNodes
+        if (filters.length < 1) {
+          return "None"
+        }
+        // Generate comma-separated list of all current filters
+        filterString += filters[0].id
```

Comment:
Array.join() will convert an array into a string, with comma as default delimiter.
Since you are getting filter[x].id, you can't directly use .join() - but you can use a lambda to call .id before joining them. Like:
filters.map(filter => filter.id).join()